### ISSUE TITLE
EasyDIY OpenSource Ebike display compatibility

### ISSUE THREAD

**Issue by dodox1** [anszom/Color_LCD#3]
Hello,
do You think it is complicated to modify Your SW102 code to OpenSource display (based on nordic nrf52 usb dongle)?

Thank's
dodo

**Comment by anszom** [comment]
No, it should be quite simple. You would need to build for a different SOC (the DIY display uses nrf52840 instead of 51822), and adjust the pin numbers. Other than that I don't think there would be any major changes.

One thing to note is that the UI uses all four buttons (up/down/M/power), so you should build the 4-button version of the display.

**Comment by dodox1** [comment]
It seems to be necessary to update the SDK
"SDK v12.3.0 provides experimental support for the nRF52840 SoC"
*** Scope for the nRF52840 chip
********************************
The PCA10056 development board is supported but all examples and libraries 
for the new chip should be treated as experimental.
  
The following SDK features are supported on the new nRF52840 chip:
  - Most BLE, hardware peripheral, and NFC examples (with some
    exceptions; see the available example projects for details)
  - Peripheral HAL and drivers (both for existing and new peripherals)
  - Library supporting CC310 CryptoCell
  - NFC Type 2 Tag and Type 4 Tag
 
The following SDK features are not supported on the new nRF52840 chip:
   - ANT
   - Secure DFU
   - Serialization
   - DTM
   - ESB and Gazell
   - RTX and FreeRTOS
   - Eddystone

**Comment by dodox1** [comment]
Just now, I found this...
https://github.com/OpenSourceEBike/Color_LCD/blob/master/firmware/SW102/Makefile_NRF52840

**Comment by dodox1** [comment]
Update. 
It seems that nrf52840 is not a big problem to convert to the newer SDK (16). (Just a few modifications of some changed API calls)
For testing purposes, I did a nasty merge of your code with the "ev_display_bluetooth_ant" source.
However, what I found a bit tricky is to convert the display driver part of your code to the SPI ssd1306 display library, which is used in the ev_display code.
